### PR TITLE
document test suite dependency on ShellCheck

### PR DESCRIPTION
**Pull request by reidpr**
Addresses #645.

I really wanted to put it in the "recommended tests with tarballs" section, but it seemed too much. Opinions?

### CODE REVIEW

File: configure.ac [hunk 1]
```diff
@@ -306,6 +306,13 @@ EOF
   AC_MSG_RESULT($have_requests)
 ])
 
+# ShellCheck
+vmin_shellcheck=0.6.0
+AC_CHECK_PROG([SHELLCHECK], [shellcheck], [shellcheck])
+# https://stackoverflow.com/questions/6022384
+CH_CHECK_VERSION([SHELLCHECK], [$vmin_shellcheck],
+                 [--version | sed -n '2{s/^version: //;p;q}'])
```
Comment (rstyd) [inline configure.ac:314]: Can you explain this sed command?

Comment (reidpr) [reply configure.ac:314]: Yes; all the `CH_CHECK_VERSION` calls have something similar, to pull out a stand-alone version number without all the clutter from bare `--version`. If you're asking what the `-n` and `p;q` does, that *should* be explained in the SO link but I can clarify if needed.

File: configure.ac [hunk 2]
```diff
@@ -610,5 +618,6 @@ Test suite
   complete test suite: ${have_tests_all}
     recommended tests with SquashFS ... ${have_tests_squash}
     documentation ... ${have_docs}
+    ShellCheck ≥ $vmin_shellcheck ... ${SHELLCHECK_VERSION_NOTE}
```
Comment (j-ogas) [inline configure.ac:621]: Where is `$SHELLCHECK_VERSION_NOTE` being defined?

File: configure.ac (hunk 1 above)
Comment (rstyd) [reply configure.ac:314]: Ah. Perfect!

File: configure.ac (hunk 2 above)
Comment (reidpr) [reply configure.ac:621]: Sorry, my comment got eaten. It is part of the magic of `CH_CHECK_VERSION`.

**Pull request merged**